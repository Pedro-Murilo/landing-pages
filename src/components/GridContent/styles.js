import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.sizes.huge} 0;
  `}
`;
