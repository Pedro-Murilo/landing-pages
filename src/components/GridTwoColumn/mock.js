export default {
  title: 'Grid Two Column',
  text: 'Lorem',
  srcImg: 'assets/images/logo.svg',
};
