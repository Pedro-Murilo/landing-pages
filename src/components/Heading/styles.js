import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.xmedium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkTheme, size, uppercase }) => css`
    color: ${darkTheme ? theme.colors.red : theme.colors.green};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
