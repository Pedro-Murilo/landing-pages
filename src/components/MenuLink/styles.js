import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacing.sizes.small};
    color: ${theme.colors.green};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.red};
      transition: all 0.3s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
