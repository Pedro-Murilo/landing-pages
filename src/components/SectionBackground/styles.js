import styled, { css } from 'styled-components';

const isBackgroundActive = (theme) => css`
  background: ${theme.colors.black};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.black};
    ${background && isBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
