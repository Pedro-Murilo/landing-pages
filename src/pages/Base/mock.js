import linksMock from '../../components/NavLink/mock';

export default {
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Test footer</p>',
};
