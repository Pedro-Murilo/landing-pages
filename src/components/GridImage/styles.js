import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacing.sizes.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.sizes.xmedium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 200ms ease-in;

    &:hover {
      transform: scale(1.1) rotate(5deg);
    }
  `}
`;
