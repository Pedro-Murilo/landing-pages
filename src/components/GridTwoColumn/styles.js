import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.sizes.medium};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacing.sizes.xmedium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.sizes.medium};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
