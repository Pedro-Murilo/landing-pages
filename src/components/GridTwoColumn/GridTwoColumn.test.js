import { renderTheme } from '../../styles/RenderTheme';
import { screen } from '@testing-library/react';

import { GridTwoColumn } from '.';

import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
