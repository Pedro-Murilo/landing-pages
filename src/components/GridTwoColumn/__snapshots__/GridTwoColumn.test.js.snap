// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid 1`] = `
.c5 {
  color: #e34040;
  font-size: 5.6rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4rem;
}

.c0 {
  background: #f7f5f5;
  color: #252525;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Grid Two Column
          </h1>
          <p
            class="c6"
          >
            Lorem
          </p>
        </div>
      </div>
      <div
        class=""
      >
        <img
          alt="Grid Two Column"
          class=""
          src="assets/images/logo.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
